include/envoy/http/metadata_interface.h

```diff
@@ -22,10 +21,13 @@ namespace Http2 {
 const uint8_t METADATA_FRAME_TYPE = 0x4d;
 const uint8_t END_METADATA_FLAG = 0x4;
 
+// NGHTTP2_MAX_PAYLOADLEN in nghttp2.
+const uint64_t METADATA_MAX_PAYLOAD_SIZE = 16384;
```

Comment:
As a sanity check, if nghttp2 changed their NGHTTP2_MAX_PAYLOADLEN to be lower, they'd reject large data chunks and your large metadata frame test would fail when we imported the change, right?

Reply:
Great question! I increase METADATA_MAX_PAYLOAD_SIZE to 20000, and the test fails at:
source/common/http/http2/codec_impl.cc:623
ASSERT(len >= size_to_copy);

Added a comment to check NGHTTP2_MAX_PAYLOADLEN in case of the assert failure.

Reply:
Make this as constexpr (perhaps for all constants), and do `static_assert` against `NGHTTP2_MAX_PAYLOADLEN`? It will fail at compile time then.